### ISSUE TITLE
Unable to build on Windows after cygwin update

### ISSUE THREAD

**Issue by ghost** [openwall/john#5324]
Recently `cygwin` has been updated [1] to:
```
Cygwin 3.4.7
gcc (GCC) 11.4.0
```

Since then we cannot build on Windows:
``` 
configure: error: cannot compute sizeof (size_t)
``` 

The root cause is related to #1093's workaround (default being C89). Maybe also #5297.

Anyway, there is no proposed fix for now (other than removing the C89 flag):
* JtR will have to deprecate some things sooner or later. Probably sooner rather than later.

*******
[1] From:
```
Cygwin 3.4.6
gcc (GCC) 11.3.0
```

**Comment by ghost** [comment]
I reverted https://github.com/openwall/john/commit/f310326a79dc869b0db1329675b1cd072798e436 and it works (Ubuntu LTS):

As seen at https://github.com/openwall/john/commit/f310326a79dc869b0db1329675b1cd072798e436#commitcomment-12118858, why do we keep '-std=gnu89`?
****

Well, other workaround is:
```diff
diff --git a/src/configure b/src/configure
index f159349b5..88198503e 100755
--- a/src/configure
+++ b/src/configure
@@ -6901,8 +6901,8 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 fi
       if test 1 -gt 0; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports -std=gnu89" >&5
-$as_echo_n "checking if $CC supports -std=gnu89... " >&6; }
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports -std=gnu99" >&5
+$as_echo_n "checking if $CC supports -std=gnu99... " >&6; }
 fi
   ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'

[...]
```

```sh
[...]
checking if gcc supports -std=gnu99... yes
[...]
Version: 1.9.0-jumbo-1+bleeding-c57beb2733 2023-06-22 08:12:50 -0300
Build: cygwin 64-bit x86_64 AVX2 AC OMP
SIMD: AVX2, interleaving: MD4:3 MD5:3 SHA1:1 SHA256:1 SHA512:1
CPU tests: AVX2
$JOHN is /run/
Format interface version: 14
Max. number of reported tunable costs: 4
Rec file version: REC4
Charset file version: CHR3
CHARSET_MIN: 1 (0x01)
CHARSET_MAX: 255 (0xff)
CHARSET_LENGTH: 24
SALT_HASH_SIZE: 1048576
SINGLE_IDX_MAX: 32768
SINGLE_BUF_MAX: 4294967295
Effective limit: Max. KPC 32768
Max. Markov mode level: 400
Max. Markov mode password length: 30
gcc version: 11.4.0
Crypto library: OpenSSL
OpenSSL library version: 01010115f
OpenSSL 1.1.1u  30 May 2023
GMP library version: 6.2.1
File locking: fcntl()
fseek(): fseek
ftell(): ftell
fopen(): fopen
memmem(): System's
times(2) sysconf(_SC_CLK_TCK) is 1000
Using times(2) for timers, resolution 1 ms
HR timer: QueryPerformanceCounter(), latency 100 ns
Total physical host memory: 7167 MiB
Available physical host memory: 5312 MiB
Terminal locale string: C.UTF-8
Parsed terminal locale: UTF-8
```

```sh
Will run 2 OpenMP threads
Testing: descrypt, traditional crypt(3) [DES 256/256 AVX2]... (2xOMP) PASS
Testing: bsdicrypt, BSDI crypt(3) ("_J9..", 725 iterations) [DES 256/256 AVX2]... (2xOMP) PASS
Testing: md5crypt, crypt(3) $1$ (and variants) [MD5 256/256 AVX2 8x3]... (2xOMP) PASS
Testing: md5crypt-long, crypt(3) $1$ (and variants) [MD5 32/64]... (2xOMP) PASS
Testing: bcrypt ("$2a$05", 32 iterations) [Blowfish 32/64 X3]... (2xOMP) PASS
[...]
Testing: dynamic_1013 [md5($p.PMD5(username)) (webEdition CMS) 256/256 AVX2 8x3]... PASS
[...]
Testing: crypt, generic crypt(3) [?/64]... (2xOMP) PASS
All 403 formats passed self-tests!
```

**Comment by solardiz** [comment]
I'm surprised we still have that `-std=gnu89` - we're using C99 already, but perhaps those specific C99 features that we use were also GNU extensions before? Anyway, I think we should try reverting that commit like Claudio has already tied on Cygwin, and should retest that across a range of gcc and clang versions. We have gcc 5.x, that was previously found problematic, as one of the many versions installed on `super` - so should try building with it there.

**Comment by ghost** [comment]
Fixed but one note remains (see #5326).
